src/datasources/asset/AssetDataSource.ts

```diff
@@ -98,6 +99,22 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
     }
   }
 
+  async metricFindQuery({ minionIds, workspace }: AssetQuery): Promise<MetricFindValue[]> {
+    const conditions = [];
+    if (minionIds?.length) {
+      minionIds = replaceVariables(minionIds, this.templateSrv);
+      const systemsCondition = minionIds.map(id => `${AssetFilterProperties.LocationMinionId} = "${id}"`);
+      conditions.push(`(${systemsCondition.join(' or ')})`);
+    }
+    workspace = this.templateSrv.replace(workspace);
+    if (workspace) {
+      conditions.push(`workspace = "${workspace}"`);
+    }
+    const assetFilter = conditions.join(' and ');
+    const assetsResponse: AssetModel[] = await this.queryAssets(assetFilter, 1000);
```

Comment:
I think this we need to specify in the HLD and discuss with Josh. Should we query for 1000 assets here ? We also don't have projection here so it will eat up a lot of unnecessary resources both on client and on the backend